Common++/header/PointerVector.h

```diff
@@ -195,6 +257,29 @@ namespace pcpp
 			return result;
 		}
 
+		/**
+		 * Removes an element from the vector and transfers ownership to the returned unique pointer.
+		 * @param[in] index The index of the element to detach.
+		 * @return An unique pointer that holds ownership of the detached element.
+		 */
+		std::unique_ptr<T> getAndDetach(size_t index)
+		{
+			return getAndDetach(m_Vector.begin() + index);
+		}
+
+		/**
+		 * Removes an element from the vector and transfers ownership to the returned unique pointer.
+		 * @param[in, out] position An iterator pointing to the element to detach.
+		 *	The iterator is shifted to the following element after the detach completes.
+		 * @return An unique pointer that holds ownership of the detached element.
+		 */
+		std::unique_ptr<T> getAndDetach(VectorIterator& position)
```

Comment:
I wonder if it is necessary to return the new `position`, I think the user will only care about the return value, so maybe `const ectorIterator&`?

Reply:
sure.